core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -376,43 +383,59 @@ public Path getWorkingDirectory() {
   /**
    * {@inheritDoc}
    *
-   * Sets up a lazy connection to Alluxio through mFileSystem.
+   * Sets up a lazy connection to Alluxio through mFileSystem. This method will override and
+   * invalidate the current contexts. This must be called before client operations in order to
+   * guarantee the integrity of the contexts, meaning users should not alternate between using the
+   * Hadoop compatible API and native Alluxio API in the same process.
+   *
+   * Initialize is guaranteed to only be called once per process.
    */
+  @SuppressFBWarnings("ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD")
```

Comment:
Q: is it possible to move `sInitialized = true;` to a static method and call this static method within the synchronized block to eliminate this warning?
just some thought, not a request